multipaz/src/commonMain/kotlin/org/multipaz/asn1/ASN1Integer.kt

```diff
@@ -42,6 +44,23 @@ class ASN1Integer(
         fun parse(content: ByteArray, tag: Int): ASN1Integer {
             return ASN1Integer(content, tag)
         }
+
+        /**
+         * Generates a [ASN1Integer] with [numBits] bits of random
+         *
+         * @param numBits number of bits of random, must be positive and divisible by 8.
+         * @param random the [Random] to used for randomness.
+         * @return a [ASN1Integer]
+         */
+        fun fromRandom(
+            numBits: Int,
+            random: Random = Random.Default
+        ): ASN1Integer {
+            require(numBits >= 8 && numBits.and(0x07) == 0) {
+                "numBits must be positive and a multiple of 8"
+            }
+            return ASN1Integer(byteArrayOf(random.nextInt(1, 127).toByte()) + random.nextBytes(numBits/8 - 1))
```

Comment:
Shouldn't this be random.nextInt(0, 128)?
The upper value is exclusive, not inclusive, so the existing call returns an int form 1 to 126.
And is there a reason we wouldn't allow 0?

Reply:
Nice catch, random.nextInt(x, y) is indeed inclusive x exclusive y.

Reading the encoding rules one more time (see https://www.itu.int/ITU-T/studygroups/com17/languages/X.690-0207.pdf), I think that we're actually good if we use `Random.nextInt(1, 255)` this way we satisfy this

```
8.3.2 If the contents octets of an integer value encoding consist of more than one octet, then the bits of the first octet
and bit 8 of the second octet:
a) shall not all be ones; and
b) shall not all be zero. 
```

and at the same time we generate all the bits requested (almost, close enough).